### PR TITLE
feature shenyu-plugin:add shenyu-plugin-logging-huawei-lts

### LINKED ISSUE #4829: [Task] HuaweiCloud Lts log plugin
### Description

i will develop huaweicloud lts log plugin

### Task List

1.huawei cloud lts plugin code
2.huawei cloud lts plugin test cases
3.huawei cloud lts plugin document, document template refer:https://github.com/apache/shenyu-website/issues/614

### PR DESCRIPTION
**Pull request by xcsnx**
Fixes #4829

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.shenyu</groupId>
+            <artifactId>shenyu-plugin-logging-common</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.github.huaweicloud</groupId>
+            <artifactId>lts-sdk-java</artifactId>
+            <version>1.0.1</version>
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/pom.xml:46]: use common properties

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        this.projectId = huaweiLtsLogConfig.getProjectId();
+        this.logGroupId = huaweiLtsLogConfig.getLogGroupId();
+        this.logStreamId = huaweiLtsLogConfig.getLogStreamId();
+        if (StringUtils.isBlank(accessKeyId) || StringUtils.isBlank(accessKeySecret) || StringUtils.isBlank(projectId)
+                || StringUtils.isBlank(regionName) || StringUtils.isBlank(logGroupId) || StringUtils.isBlank(logStreamId)) {
+            LOG.error("init Huawei lts client error, please check projectId, accessKeyId, accessKeySecret, regionName, logGroupId or logStreamId");
+            return;
+        }
+        JavaSDKAppender appender = JavaSDKAppender.custom()
+                // 华为云帐号的项目ID（project id）
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java:76]: use english comment

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                .setBatchSizeThresholdInBytes(huaweiLtsLogConfig.getBatchSizeThresholdInBytes())
+                // producer发送单批日志条数上限
+                .setBatchCountThreshold(huaweiLtsLogConfig.getBatchCountThreshold())
+                // producer发送单批日志等待时间
+                .setLingerMs(huaweiLtsLogConfig.getLingerMs())
+                // producer发送日志失败后重试次数
+                .setRetries(huaweiLtsLogConfig.getRetries())
+                // 首次重试的退避时间
+                .setBaseRetryBackoffMs(huaweiLtsLogConfig.getBaseRetryBackoffMs())
+                // 重试的最大退避时间
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java:98]: same

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        contents.add(logContent);
+
+        logItem.setContents(contents);
+        logItemList.add(logItem);
+
+        try {
+            final ListenableFuture<Result> f = producer.send(logGroupId, logStreamId, logItemList);
+            Futures.addCallback(f, new ProducerFutureCallback(logGroupId, logStreamId), threadExecutor);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java:152]: not throw runtime exception

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/plugin/huawei/lts/handler/LoggingHuaweiLtsPluginDataHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private final PluginData pluginData = new PluginData();
+
+    @BeforeEach
+    private void setUp() {
+        this.loggingHuaweiLtsPluginDataHandler = new LoggingHuaweiLtsPluginDataHandler();
+        pluginData.setEnabled(true);
+        pluginData.setConfig("{\n"
+                + "    \"projectId\": \"3d696d796b9647f982b5ec787b2ca249\",\n"
+                + "    \"logGroupId\": \"8285fbe3-9c73-4deb-a8d0-96ec06c4af3a\",\n"
+                + "    \"logStreamId\": \"8b4fe176-4a7e-4bd5-8bb7-17d0ec6730b2\",\n"
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/plugin/huawei/lts/handler/LoggingHuaweiLtsPluginDataHandlerTest.java:42]: use your accessId and secret key

**Review by moremind — APPROVED** [review]

File: shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                .setAccessKeySecret(accessKeySecret)
+                // 云日志服务的区域
+                .setRegionName(regionName)
+                // 单个Appender能缓存的日志大小上限
+                .setTotalSizeInBytes(huaweiLtsLogConfig.getTotalSizeInBytes())
+                // producer发送日志时阻塞时间
+                .setMaxBlockMs(huaweiLtsLogConfig.getMaxBlockMs())
+                // producer发送单批日志量上限
+                .setBatchSizeThresholdInBytes(huaweiLtsLogConfig.getBatchSizeThresholdInBytes())
+                // producer发送单批日志条数上限
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-logging/shenyu-plugin-logging-huawei-lts/src/main/java/org/apache/shenyu/plugin/huawei/lts/client/HuaweiLtsLogCollectClient.java:90]: not comment with .

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/springboot/starter/plugin/logging/huawei/lts/LoggingTencentClsPluginConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@Configuration
+@EnableConfigurationProperties
+public class LoggingTencentClsPluginConfigurationTest {
```
Comment (moremind) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/springboot/starter/plugin/logging/huawei/lts/LoggingTencentClsPluginConfigurationTest.java:33]: cls？

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/springboot/starter/plugin/logging/huawei/lts/LoggingTencentClsPluginConfigurationTest.java (hunk 7 above)
Comment (xcsnx) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-logging-huawei-lts/src/test/java/org/apache/shenyu/springboot/starter/plugin/logging/huawei/lts/LoggingTencentClsPluginConfigurationTest.java]: Thanks, I will improve the code

**Comment by yunlongn** [comment]
pls add data.  -> `./db/upgrade/2.5.1-upgrade-2.6.0-*.sql`

**Comment by xcsnx** [comment]
> pls add data. -> `./db/upgrade/2.5.1-upgrade-2.6.0-*.sql` @xcsnx
ok

**Review by moremind — APPROVED** [review]

**Pull request merged**